javascript-modules-library/src/core/server/utils/urlBuilder/urlBuilder.ts

```diff
@@ -165,7 +156,10 @@ export function buildUrl(
       props.parameters &&
       Object.prototype.toString.call(props.parameters) === "[object Object]"
     ) {
-      url = appendParameters(url, props.parameters);
+      const urlWithParams = new URL(url);
```

Comment:
You might have an issue here for relative urls, URL only supports fully resolved urls